backend/src/apiserver/client_manager.go

```diff
@@ -374,3 +377,35 @@ func initPipelineVersionsFromPipelines(db *gorm.DB) {
 
 	tx.Commit()
 }
+
+func backfillExperimentIDToRunTable(db *gorm.DB) (retError error) {
+	// check if there is any row in the run table has experiment ID being empty
+	var emptyExperimentUUID = false
+	rows, err := db.CommonDB().Query(`SELECT ExperimentUUID FROM run_details WHERE ExperimentUUID = '' LIMIT 1`)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for !emptyExperimentUUID && rows.Next() {
```

Comment:
looks like you don't need emptyExperimentUUID. Can you just do 
if ! row.Next()  {
 return nil
}